From 916e1c5bfbb1364697bbe4f4d4e86657568c44c1 Mon Sep 17 00:00:00 2001
From: Martin Liska <martin.liska@hey.com>
Date: Wed, 24 Jan 2024 13:29:19 +0100
Subject: [PATCH] Fix LLVM 18 build error:

/__w/cvise/cvise/clang_delta/CommonRenameClassRewriteVisitor.h:392:3: error: enumeration value 'StructuralValue' not handled in switch [-Werror=switch]
  392 |   switch (Arg.getKind()) {
      |   ^~~~~~
---
 clang_delta/CommonRenameClassRewriteVisitor.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/clang_delta/CommonRenameClassRewriteVisitor.h b/clang_delta/CommonRenameClassRewriteVisitor.h
index 90a3144f..f2daeb24 100644
--- a/clang_delta/CommonRenameClassRewriteVisitor.h
+++ b/clang_delta/CommonRenameClassRewriteVisitor.h
@@ -394,6 +394,9 @@ bool CommonRenameClassRewriteVisitor<T>::TraverseTemplateArgumentLoc(
   case TemplateArgument::Null:
   case TemplateArgument::Declaration:
   case TemplateArgument::Integral:
+#if LLVM_VERSION_MAJOR >= 18
+  case TemplateArgument::StructuralValue:
+#endif
     return true;
 
   case TemplateArgument::Type: {
